test-framework/core/src/main/java/org/keycloak/testframework/https/ManagedCertificates.java

```diff
@@ -29,104 +30,167 @@ public class ManagedCertificates {
-        clientsTruststorePath = KEYSTORES_DIR.resolve("kc-testing-clients-truststore" + "." + configBuilder.getKeystoreFormat().getPrimaryExtension());
+        keystoreFormat = configBuilder.getKeystoreFormat();
+        mTlsEnabled = configBuilder.isMTlsEnabled();
+        password = keystoreFormat == KeystoreUtil.KeystoreFormat.JKS ? "password".toCharArray() : "passwordpassword".toCharArray();
+
+        initServerAndClientCertificateObjects();
+
+        initKeyAndTrustStores();
+    }
+
+    public void initServerAndClientCertificateObjects() {
+        Path serverKeystorePath = resolvePath("kc-testing-server-keystore");
+        Path serverTruststorePath = resolvePath("kc-testing-server-truststore");
+        server = new Certificate(serverKeystorePath, serverTruststorePath);
+        server.setKeyEntry("server-private-key");
+        server.setCertificateEntry("server-certificate");
+
+        if (mTlsEnabled) {
+            Path clientKeystorePath = resolvePath("kc-testing-client-keystore");
+            Path clientTruststorePath = resolvePath("kc-testing-client-truststore");
+            client = new Certificate(clientKeystorePath, clientTruststorePath);
+            client.setKeyEntry("client-private-key");
+            client.setCertificateEntry("client-certificate");
+        }
+    }
+
+    private Path resolvePath(String fileName) {
+        return KEYSTORES_DIR.resolve(fileName + "." + keystoreFormat.getPrimaryExtension());
+    }
 
-        password = configBuilder.getKeystoreFormat() == KeystoreUtil.KeystoreFormat.JKS ? "password".toCharArray() : "passwordpassword".toCharArray();
+    public SSLContext getSSLContext() {
+        try {
+            if (mTlsEnabled) {
+                return SSLContextBuilder.create()
+                        .loadTrustMaterial(client.getTruststorePath().toFile(), password)
+                        .loadKeyMaterial(client.getKeystore(), password)
+                        .build();
+            } else {
+                return SSLContextBuilder.create()
+                        .loadTrustMaterial(server.getTruststore(), null)
+                        .build();
+            }
+        } catch (KeyManagementException | NoSuchAlgorithmException | KeyStoreException e) {
+            throw new ManagedCertificatesException(e);
+        } catch (UnrecoverableKeyException | CertificateException | IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
 
-        initServerCerts(configBuilder.getKeystoreFormat());
+    public String getServerKeyStorePath() {
+        return server.getKeystorePath().toString();
     }
 
-    public String getKeycloakServerKeyStorePath() {
-        return serverKeystorePath.toString();
+    public String getServerTrustStorePath() {
+        return server.getTruststorePath().toString();
     }
 
-    public String getKeycloakServerKeyStorePassword() {
+    public String getKeyAndTrustStorePassword() {
```

Comment:
I'd split this into 2; `getServerKeyStorePassword` and `getServerTrustStorePassword`. It's fine they are using the same password now, but could be different in the future, or for a different supplier.